File: PRACTICAS/practica3/Civitas_P1/src/civitas_p1/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.Random;

/**
 *
 * @author angel
 */
public class Dado {
    
    private Random random;
    private int ultimoResultado;
    private Boolean debug;
    
    public static final Dado instance = new Dado();
    private static int SalidaCarcel = 5;
    
    Dado(){
        
        debug = false;
        ultimoResultado = 0;
        random = new Random();
        
    }
    
    static public Dado getInstance(){
       return instance;
    }
   
    int tirar(){
       
       if (!debug)
           ultimoResultado = random.nextInt(6) + 1;
       
       else
           ultimoResultado = 1;
       
       return ultimoResultado;       
       
    }
   
   Boolean salgoDeLaCarcel(){
       return tirar() >= 5;
   }
   
   int quienEmpieza(int n){
       return random.nextInt(n);
   }
   
   public void setDebug(Boolean d){
       
       debug = d;
       
       if (d)
           Diario.instance.ocurreEvento("El debug se ha puesto a true");
       else
           Diario.instance.ocurreEvento("El debug se ha puesto a false");
       
   }
   
   int getUltimoResultado(){
       return ultimoResultado;
   }   
    
}

File: PRACTICAS/practica1/Civitas_Ruby/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require 'singleton'
module Civitas

class Dado
  
  include Singleton
  
  #Atributo de clase
  @@SalidaCarcel = 5
  
  
  
  #Constructor privado sin argumentos
  def initialize
    
    @debug = false
    @ulimoresultado = 0
    @random = rand(6) + 1
    
  end
  
  #Metodos de instancia
  def tirar
    
    if (@debug==false)
      
      @ultimoresultado = rand (1..6)
      return @ultimoresultado
      
    else
    
      return 1
      
    end
    
  end
  
  
  def salgodelacarcel
    
    @resultado = tirar
    
    if (@resultado >= 5)
      @sale = true
    else
      @sale = false
    end
    
    return @sale
    
  end
  
  
  
  def quienempieza(n)
    
    return rand (n)
    
  end
  
  
  def setdebug (d)
    
    if(d == true)
      @debug = true
      Diario.instance.ocurre_evento("Debug mode activado")
    else
      @debug = false
      Diario.instance.ocurre_evento("Debug mode desactivado")
    end
    
  end
  
  
  def getultimoresultado
    
    return @ultimoresultado
    
  end
  
  
  
  
end
end